framework/src/times/ControllableInputTimes.C

```diff
@@ -0,0 +1,48 @@
+//* This file is part of the MOOSE framework
+//* https://mooseframework.inl.gov
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ControllableInputTimes.h"
+
+registerMooseObject("MooseApp", ControllableInputTimes);
+
+InputParameters
+ControllableInputTimes::validParams()
+{
+  InputParameters params = InputTimes::validParams();
+  params.addClassDescription("Times set directly from a user parameter in the input file");
+  params.addRequiredParam<Real>("next_time", "Time to store in the times vector");
+  params.declareControllable("next_time");
+
+  // Times are known for all processes already
+  params.set<bool>("auto_broadcast") = false;
+  params.makeParamNotRequired("times");
+
+  return params;
+}
+
+ControllableInputTimes::ControllableInputTimes(const InputParameters & parameters)
+  : InputTimes(parameters), _next_time(getParam<Real>("next_time"))
+{
+  // Initialize times array
+  _times.push_back(0);
```

Comment:
I really dont want this. Same as in FunctorTimes

Not all simulations start at time = 0. So at minimum we want it to execution.startTimes

Reply:
This will be updated at initialization step with a real one. I'll also do sort of the time array. I think initialization with zero doesn't affect anything(no case breaks)

Reply:
> Plus you can't have execution.startTimes available during construction phase, can you

That's just another reason not to do it then. 

> So far we have no issues with zero initialization

Right so we dont want to add an issue with a 0 that is not necessary in the Times vector

Reply:
I don't have other solutions for these. The only thing I could think of is to change the size check of _input_times to allow empty array

Reply:
That is preferable if that's the only thing we can do